common_code/inspiral/CE_orbit.f90

```diff
@@ -304,7 +460,273 @@ subroutine calc_quantities_at_comp_position(id, ierr)
+         ! Relative velocity
+         magv =  s% xtra(16) / Rsun * secyer !sqrt( f(1)*f(1) + r*r*f(3)*f(3) )
+
+         ! r and nu
+         r =  y(1)
+         nu = y(3)
+
+         ! dr/dt and dnu/dt
+         f(1) = y(2)             !dr/dt
+         f(3) = y(4)             !dnu/dt
+
+         
+
+         arr = s% r
+         value = r * Rsun
+         ! Call the subroutine to find the index at the location of the companion
+         CALL FindApproximateValueIndex(arr, value, index)
+         DEALLOCATE(arr)
+
+         
+         ! Eclosed mass at the position of the companion
+         M = s% m(index) / Msun
+
+         ! Gravitational parameter 
+         mu = G * ( CE_companion_mass + M )
+
+         ! External force and its components
+         renv = s% r(1) / Rsun
+        
+
+         if (r.gt.renv) then
+               Fext = 0.d0
+         else
+            if (s% x_integer_ctrl(2) .eq. 4) then
+               ! Drag force as described in Ginat+(2020)
+               ! from Ostriker (1999) and Binney & Tremaine (2008).
+               tmprho = s% rho(index) / Msun * (Rsun**3.d0)
+               tmp1 = CE_companion_mass * M / (CE_companion_mass + M)
+               tmp2 = 4.d0 * pi * G*G  * tmp1 * tmprho
+               
+               if (machn.lt. 1.d0)  then
+                  tmp3 = log( (1.d0 + machn)/(1.d0 - machn) * exp(-2.d0 * machn) )
+               else
+                  bmax = 2.d0 * r
+                  bmin = R_acc / Rsun !2.d0 * G*mcomp / (magv*magv + (csound(index) * rsunyr)*(csound(index) * rsunyr))
+                  lambda = bmax / bmin
+                  tmp3 = log(lambda*lambda - (lambda*lambda / (machn*machn)))
+               endif
+               ! must be in units of Rsun / yr^2
+               Fext = - (tmp2 * tmp3 ) / (magv*magv) ! this should be in units of acceleration
+               !Fext = 0.d0
+               ! Adding hydrodynamical drag
+               Fext = Fext - pi * CE_companion_radius**2.d0 * tmprho * magv**2 / CE_companion_mass
+            else
+               stop "Select x_integer_ctrl(2)=4 as the complete orbit integration is only defined for this option"
+            end if 
+
+         endif
+
+         ! Saving the magnitude of the drag force (Msun Rsun / yr^2)
+         s% xtra(30) = abs(Fext) * CE_companion_mass
+
+
+         
+         Fr  = Fext * (f(1) / magv)
+         Fnu = Fext * (r * f(3) / magv) ! be careful with the units
+
+         ! 2.5 Post Newtonian terms
+         PN25 = (magv/c0)*(magv/c0) + 1.5d0 * (magv/c0)*(magv/c0)*(magv/c0)*(magv/c0)
+
+
+
+
+
+         ! d2r/dt2 and d2nu/dt2
+         f(2) = - mu / ( y(1)*y(1) ) * (1.0 + PN25) + Fr + r * f(3)*f(3) ! d2r/dt2
+
+         f(4) = Fnu / r - (2.d0 * f(1) * f(3) / r)                        ! d2nu/dt2
+
+      
+
+      end subroutine kepler_polar
+
+
+
+      
+! ***********************************************************************
+      SUBROUTINE FindApproximateValueIndex(arr, value, index)
```

Comment:
Do we need this? I think we should get all values as interpolates ones from `calc_quantities_at_comp_position`.